git/index/base.py

```diff
@@ -1210,9 +1210,9 @@ def checkout(
         def handle_stderr(proc: "Popen[bytes]", iter_checked_out_files: Iterable[PathLike]) -> None:
             stderr_IO = proc.stderr
             if not stderr_IO:
-                return None  # Return early if stderr empty.
-            else:
```

Comment:
This took me a while to understand and required side-by-side diffing to actually be obvious. I absolutely wasn't aware empty else: blocks are allowed and even snuck in here.

Reply:
Empty `else` blocks are not allowed; that would be a syntax error. This was an `else` with the statement `stderr_bytes = stderr_IO.read()`. But that statement didn't need to go in an `else`, because the code under the `if` was a `return` statement.

(I've temporarily unresolved this conversation to make the message visible, since I'm not sure it otherwise would be. But this can be marked resolved again at any time, if desired.)

Reply:
Thanks for the clarification, it all makes sense now :).